src/core/modules/impl/handler/JsHandlerApi.ts

```diff
@@ -49,6 +49,7 @@ export class JsHandlerApi<State> extends AbstractContractHandler<State> {
       const handlerResult = await Promise.race([timeoutPromise, this.contractFunction(stateCopy, interaction)]);
 
       if (handlerResult && (handlerResult.state !== undefined || handlerResult.result !== undefined)) {
+        await this.swGlobal.kv.commit();
```

Comment:
`commit` calls db.batch with an array of all operations 'registered' during interaction evaluation.